ERIC.js

```diff
@@ -150,6 +154,75 @@ async function scrape(doc, url = doc.location.href) {
 	await em.doWeb(doc, url);
 }
 
+function cleanInput(search) {
+	if (typeof search === 'string') {
+		search = { ericNumber: search };
+	}
+	if (typeof search.ericNumber !== 'string') {
+		return false;
+	}
+	let matches = search.ericNumber.match(/E[DJ]\d+/);
+	if (matches) {
+		search.ericNumber = matches[0];
+		return search;
+	}
+	else {
+		return false;
+	}
+}
+
+function detectSearch(search) {
+	return !!cleanInput(search);
+}
+
+async function doSearch(search) {
+	search = cleanInput(search);
+	let { response } = await requestJSON(`https://api.ies.ed.gov/eric/?search=id:${search.ericNumber}&format=json&fields=*`);
+	if (!response.docs || !response.docs.length) {
+		throw new Error('ERIC search returned no results');
+	}
+	let doc = response.docs[0];
+	let item = new Zotero.Item(getType(doc.id, doc.publicationtype.join('; ')));
+	item.title = ZU.unescapeHTML(doc.title);
+	item.creators.push(...doc.author.map(name => ZU.cleanAuthor(name, 'author', true)));
+	item.abstractNote = doc.description || doc.desc;
+	if (item.abstractNote) {
+		item.abstractNote = ZU.unescapeHTML(item.abstractNote);
+	}
+	item.ISBN = doc.isbn && ZU.cleanISBN(doc.isbn.join(' '));
+	item.ISSN = doc.issn && ZU.cleanISSN(doc.issn.join(' ').replace(/ISSN-/g, ''));
```

Comment:
Could we try to get the DOI from the `url` field, as we do in the web scraper, see e.g. https://api.ies.ed.gov/eric/?search=id:EJ956651&format=json&fields=*